### PR TITLE
Cleanup/detector

### PR DESCRIPTION
**Pull request by abouteiller**
This PR continues the cleanup of the initialization of the prte detector.

* correct the detector infrastructure so that it does not crash during startup when compiled-in
* Correct the examples
* remove unused variables and reduce scope of locals and functions (static)

I have removed the part that makes this feature compiled-in by default, in order to get things moving.

There is a crippling bug in the detector as-is and we need this quickly.

### CODE REVIEW

File: src/mca/state/base/state_base_fns.c [hunk 1]
```diff
@@ -761,7 +761,8 @@ void prte_state_base_track_procs(int fd, short argc, void *cbdata)
             parent.jobid = jdata->jobid;
             parent.vpid = PRTE_VPID_WILDCARD;
 
-            /* do not kill send the msg if ft prte is enabled */
+            /* if ft prte is enabled, a PMIx event has already been produced
+             * and this is redundant. */
             if (!prte_enable_ft) {
```
Comment (abouteiller) [inline src/mca/state/base/state_base_fns.c:766]: prte_enable_ft is never true at the site of prun/mpiexec, even though the variable is registered and set (and is indeed set to true in prted)..

File: config/prte_configure_options.m4 [hunk 2]
```diff
@@ -408,19 +408,19 @@ PRTE_ZLIB_CONFIG
...
-    [AC_HELP_STRING([--enable-prte-ft],
-        [ENable PRRTE fault tolerance support (default: disabled)])])
-if test "$enable_prte_ft" = "yes"; then
-    AC_MSG_RESULT([yes])
-    prte_enable_ft=1
-    PRTE_SUMMARY_ADD([[Options]],[[Fault tolerance]], [prte_ft], [yes])
-else
+    [AC_HELP_STRING([--disable-prte-ft],
+        [Disable PRRTE fault tolerance support (default: enabled)])])
+if test "$enable_prte_ft" = "no"; then
```
Comment (rhc54) [inline config/prte_configure_options.m4:413]: Let's leave this `off` by default for v2.0 so groups that don't need it have one clean release they can tag. We can turn it `on` by default afterwards.

Comment (abouteiller) [reply config/prte_configure_options.m4]: In order to get things moving, I have postponed that change to a separate PR.

**Comment by jjhursey** [comment]
Related to https://github.com/open-mpi/ompi/pull/7740

File: src/mca/rml/rml_types.h [hunk 3]
```diff
@@ -185,9 +185,6 @@ BEGIN_C_DECLS
 
 /* error propagate  */
 #define PRTE_RML_TAG_RBCAST                 66
-#define PRTE_RML_TAG_BMGXCAST               67
-#define PRTE_RML_TAG_ALLGATHER_BMG          68
-#define PRTE_RML_TAG_BMG_COLL_RELEASE       69
```
Comment (gpaulsen) [inline src/mca/rml/rml_types.h:190]: Is this removal intentional?

Comment (dong0321) [reply src/mca/rml/rml_types.h:190]: Yes, we have a better way to implement the code.
